sale_quotation_capable_vendor_rfq/models/product_product.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def _get_matching_vendor_pricelists(self, quantity, days_min=0, days_max=0):
+        """
+        Returns a recordset of product.supplierinfo records that match the criteria.
+
+        :param quantity: Quantity to match
+        :param days_min: Minimum delivery time to match
+        :param days_max: Maximum delivery time to match
+        :return: Recordset of product.supplierinfo
+        """
+        self.ensure_one()
+
+        # get all vendors that match the criteria
+        supplierinfo = self.seller_ids.filtered(
+            lambda s: (not days_min or s.delay >= days_min)
+            and (not days_max or s.delay <= days_max)
+            and s.min_qty <= quantity
+            and (s.product_id == self or not s.product_id)
+        )
```

Comment:
maybe will faster:

```python
        domain = [
            ('product_tmpl_id', '=', self.product_tmpl_id.id),
            ('min_qty', '<=', quantity),
        ]
        if days_min:
            domain.append(('delay', '>=', days_min))
        if days_max:
            domain.append(('delay', '<=', days_max))
        
        supplierinfo = self.seller_ids.filtered_domain(domain)

        min_prices = {}
        for info in supplierinfo:
            if info.partner_id not in min_prices or info.price < min_prices[info.partner_id].price:
                min_prices[info.partner_id] = info
        
        return self.env['product.supplierinfo'].concat(*min_prices.values())
```

Reply:
Using the domain `('product_tmpl_id', '=', self.product_tmpl_id.id)` retrieves all supplier information associated with the product template, regardless of the specific product variant. This can lead to issues if different variants have distinct supplier information. Please correct me if I'm wrong.